src/plugins/discover/public/components/discover_grid/discover_grid.scss

```diff
@@ -30,6 +30,15 @@
   }
 }
 
+.dscDiscoverGrid__cellValue {
+  font-family: $euiCodeFontFamily;
+}
+
+.dscDiscoverGrid__cellPopoverValue {
+  font-family: $euiCodeFontFamily;
+  font-size: $euiFontSizeXS;
```

Comment:
I'd suggest `$euiFontSizeS`, which is a bit larger. I think to make it larger is fine, since the popover is here to have a closer look, which is easier with a larger font 
<img width="278" alt="Bildschirmfoto 2022-05-05 um 11 48 23" src="https://user-images.githubusercontent.com/463851/166899840-e2a9ee23-6aa7-4936-baf3-16eea9c2c224.png">